Automattic/vip-go-mu-plugins
# Fix incorrect width and height in core/image block

## Description

Addresses incorrect `img` tag `width`|`height` values in `core/image` blocks by filtering `render_block`, where it can get the attachment ID and intended size from the classes of the `img` and `figure` tags.
From there it looks for the intended size's w|h values in the attachment's postmeta, and add them to the `img` tag for the return.
Also fixes incorrect srcsets for block images without having to touch them, since it's upstream of them and supplies the values they need.
Returns original safely and silently on failure. Matches narrowly, and can operate on already-filtered blocks as long as they have the required size and ID classes.
Optional switch for `resize` args can be omitted, but it could be nice to give folks the choice. I defer to plat as to how to make this switchable (or if we should just lose it for simplicity and brevity).

## Changelog Description

### Fixed
- Fixed incorrect image sizes and srcsets for `core/image` blocks on frontend requests

## Pre-review checklist

Please make sure the items below have been covered before requesting a review:


## Pre-deploy checklist


## Steps to Test

1. Check out PR / use relevant file as generic mu-plugin
2. Examine any page/post with core/image block/s
3. Examine src of any core/image block (other than size:full)
4. Its `width` and `height` values and srcset should all be congruent with its intended size
5. It should also visually look the same size in the frontend and the editor
6. Repeat as required for any/all registered sizes
7. Optional: register some custom sizes and repeat
